src/clients/klist/klist.c

```diff
@@ -720,11 +720,6 @@ show_credential(krb5_creds *cred)
         extra_field += 2;
     }
 
-    if (extra_field > 3) {
-        fputs("\n", stdout);
-        extra_field = 0;
-    }
-
```

Comment:
This one seemed interesting, so I looked into it further.

The dead code was added in commit 3f7b5fca80c0aaeffcd9348c70613af5b63a61f1, which was scoped to adding the klist -e option.  Prior to that commit, instead of `extra_field` we had `first`, and it was just there to track whether to use a tab or ", " prefix and whether to output a newline at the end; we never stuck a newline in the middle.

We can output extra information about (1) a non-default client principal, (2) the renew-till time, (3) the flags, and (4) the etype information (after the commit).  The inoperable code would add a newline between (2) and (3) if it ever fired.

There is also a non-dead check which adds a newline between (3) and (4) if extra_field > 2.  For whatever reason, the code increments extra_field by 2 for (2) and by 1 for everything else.  (It is true that the flags display is pretty short compared to a renew-till time, but a client principal name is not.)  So that conditional triggers if (2) and at least one of (1) and (3) are present.  The net result is a fairly nice display if (2), (3), and (4) are all present:

	renew until 05/10/2019 11:02:55, Flags: FfPRAT
	Etype (skey, tkt): aes256-cts-hmac-sha1-96, aes256-cts-hmac-sha1-96 

but perhaps not so nice if only (2) or (3) are present:

	renew until 05/10/2019 11:02:55, Etype (skey, tkt): aes256-cts-hmac-sha1-96, aes256-cts-hmac-sha1-96 

	Flags: I, Etype (skey, tkt): aes256-cts-hmac-sha1-96, aes256-cts-hmac-sha1-96 

I think adding a newline between (2) and (3) was out of scope for the commit, and should generally not be necessary (non-default client principals are rare), so I will take the change as is.